Source/WebCore/Modules/notifications/NotificationJSONParser.cpp

```diff
@@ -111,18 +122,39 @@ static std::optional<unsigned long long> parseAppBadgeString(const String& badge
     });
 }
 
-ExceptionOr<NotificationPayload> NotificationJSONParser::parseNotificationPayload(const JSON::Object& object)
+ExceptionOr<NotificationPayload> NotificationJSONParser::parseNotificationPayload(const JSON::Object& outerObject)
 {
-    URL defaultActionURL;
-    if (auto value = object.getValue(defaultActionURLKey())) {
+    std::optional<int> declarativeWebPushValue;
+    if (auto value = outerObject.getValue(declarativeWebPushKey())) {
+        if (value->type() == JSON::Value::Type::Integer || value->type() == JSON::Value::Type::Double)
+            declarativeWebPushValue = value->asInteger();
+    }
+    if (!declarativeWebPushValue)
+        return Exception { ExceptionCode::NotFoundError, makeString("Push message does not have a valid '"_s, declarativeWebPushKey(), "' integer value"_s) };
+    if (*declarativeWebPushValue != declarativeWebPushValue)
```

Comment:
Is the check on L132 sufficient? Not totally understanding the point of this check

Reply:
The standard behavior for opting in to declarative web push is to have a magic key/value in the push message JSON.
That's what this check is for.

Reply:
But doesn't your declaration on L76 shadow the constant declared in L37? It seems like a non-sensical comparison to compare a value against itself.